tests/codex/blockexchange/discovery/testdiscoveryengine.nim

```diff
@@ -126,8 +127,12 @@ asyncchecksuite "Test Discovery Engine":
         want.fire()
 
     await discoveryEngine.start()
+    var idx = 0
     while peerStore.len < minPeers:
-      discoveryEngine.queueFindBlocksReq(@[blocks[0].cid])
```

Comment:
Ugh, but can't this happen in production settings too? Like, us requesting the same block more than once? Will the requests then maybe get cancelled sometimes?

Reply:
If you request discovery for a CID that's already in progress, the discovery engine will ignore it. This is the behavior we want. It's just that the test was running into this sometimes when it didn't want to.